### PR TITLE
MAGPIE_TEST_VERSION envvar

### PR DESCRIPTION
**Pull request by fmigneault**
Add test-version option to ignore safeguards during development of next version.

This is useful when a new test that brings a new feature must define `>= "future-version"` that is not yet deployed.
The tests are not run because the condition is not yet met, which always needed a fake "bump-version" to pass, and then revert for real bump/tag.

Within this, one can set `MAGPIE_TEST_VERSION = "latest"` prior to running tests and all compares to `>= "future-version"` will return `True`.

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -178,6 +178,35 @@ def __new__(cls, name, description=None):
         return make_run_option_decorator(RunOption(name, description=description))
 
 
+class TestVersion(LooseVersion):
+    """
+    Special version supporting ``latest`` keyword to ignore safeguard check of :func:`warn_version` during development.
+
+    .. seealso::
+        Environment variable ``MAGPIE_TEST_VERSION`` should be set with the desired version or ``latest`` to evaluate
+        even new features above the last tagged version.
```
Comment (ChaamC) [inline tests/utils.py:187]: Maybe the MAGPIE_TEST_VERSION could also be documented elsewhere (with an explanation similar to this comment), or added to the docker-compose.yml.example and/or magpie.env.example?

**Review by ChaamC — COMMENTED** [review]
Apart from the single comment I added, the rest seems good!

File: tests/utils.py (hunk 1 above)
Comment (fmigneault) [reply tests/utils.py:187]: Yes. I was waiting for you 'tests' section update to add it.

File: env/magpie.env.example [hunk 2]
```diff
@@ -21,10 +23,19 @@ PHOENIX_PORT=8443
 PHOENIX_PUSH=false
 TWITCHER_PROTECTED_PATH=/ows/proxy
 
-# Magpie Tests
+## === Magpie Tests ===
+# below version defines the override value for test
+# when using tes safeguards for different versions, new tests that require 'future' tags yet to be defined can
```
Comment (ChaamC) [inline env/magpie.env.example:28]: Small typo : missing 't' from test in "when using tes"

**Review by ChaamC — COMMENTED** [review]
Good improvements to the test documentation! :)

LGTM

File: env/magpie.env.example (hunk 2 above)
Comment (fmigneault) [reply env/magpie.env.example:28]: nice catch ;)

**Pull request merged**